### PR TITLE
Single-package rbi generation prototype

### PR DESCRIPTION
**Pull request by elliottt**
Initial work on adding support for single-package rbi generation for Stripe. There are some issues remaining, but parent packages name resolution is implemented which was one of the main risks for this project.

This should be relatively un-invasive: all functionality is gated on passing the `--single-package=...` flag, and the new state on `GlobalState` will default-initialize to values that disable all of the single-package functionality.

### Motivation

Working towards incremental package rbi generation for Stripe's codebase.

### Test plan

See included automated tests.

### CODE REVIEW

File: test/cli/rbi-gen-single/rbi-gen-single.sh
Comment (froydnj) [inline test/cli/rbi-gen-single/rbi-gen-single.sh:20]: This is totally non-blocking, but this script is doing very similar things to the existing single-package typechecking tests; it would be nice if they were merged somehow.  I don't know how to make that happen, exactly, because this script is checking for different things than the existing test.  But it wouldn't surprise me if we wind up needing more tests of this flavor.

File: packager/rbi_gen.cc
Comment (froydnj) [inline packager/rbi_gen.cc:1367]: I feel like there ought to be a `std::` function for this, but I'm not seeing it.

**Review by froydnj — APPROVED** [review]
This seems reasonable to me.  I have a couple comments on things that I'd like to see added before this gets merged.  Feel free to push back.

File: packager/rbi_gen.h
Comment (elliottt) [reply packager/rbi_gen.h:33]: The comment is out of date, thanks for catching this!

File: main/realmain.cc
Comment (elliottt) [reply main/realmain.cc:774]: Happy to switch over to that.

File: resolver/resolver.cc
Comment (elliottt) [reply resolver/resolver.cc:373]: Yep, it definitely will. I'll move this to an instance variable.

File: test/cli/rbi-gen-single/rbi-gen-single.sh
Comment (elliottt) [reply test/cli/rbi-gen-single/rbi-gen-single.sh:20]: I looked into merging it in, but wanted to get this PR merged with some testing present. It would be totally reasonable to rework this test into existing tests, but I'd rather not block this PR on that.

File: packager/rbi_gen.cc
Comment (elliottt) [reply packager/rbi_gen.cc:1367]: I wanted to use `std::equal`, but not being able to supply a function for the equality check breaks it. Yet another case where importing range-v3 would help :)

Comment (elliottt) [reply packager/rbi_gen.cc:1367]: It turns out that `std::equal` does take a binary predicate as of c++17, so I reworked this to use that instead.

**Pull request merged**
